crates/red_knot/src/args.rs

```diff
@@ -120,6 +133,16 @@ impl CheckCommand {
             )
         };
 
+        let mut respect_ignore_files = None;
+        // --no-respect-gitignore defaults to false and is set true by CLI flag. If passed,
+        // override config file
+        if self.no_respect_ignore_files {
+            respect_ignore_files = Some(false);
+        // Otherwise, only pass this through if explicitly set (don't default to anything here to
+        // make sure that doesn't take precedence over an explicitly-set config file value)
+        } else if self.respect_ignore_files.is_some() {
+            respect_ignore_files = Some(true);
+        }
```

Comment:
Nit: I prefer to avoid `mut` variables where possible
```suggestion
        let respect_ignore_files = self.no_respect_ignore_files.then_some(false).or(self.respect_ignore_files);
```

Reply:
great - had imagined there was a more idiomatic way to do this, thanks!